Misc clarifications and tweaks for dev container content (#5760)
microsoft/vscode-docs: docs/devcontainers/containers.md

See the [latest template](https://github.com/devcontainers/feature-template) for instructions on using the dev container CLI to publish your own public or private Features.

### Features spec and distribution

> Features are an active proposal in the open-source [Development Containers Specification](https://containers.dev). You can review [more information about how Features work](https://containers.dev/implementors/features) and their [distribution](https://containers.dev/implementors/features-distribution).

Reviewer: Remove mention of proposal here too?

```suggestion
Features are a key part of the open-source [Development Containers Specification](https://containers.dev). You can review [more information about how Features work](https://containers.dev/implementors/features) and their [distribution](https://containers.dev/implementors/features-distribution).
```

Author: Arg! Missed one!
